hazelcast-sql/src/main/java/com/hazelcast/sql/impl/calcite/opt/physical/visitor/PlanCreateVisitor.java

```diff
@@ -159,21 +164,33 @@
     /** Row metadata. */
     private SqlRowMetadata rowMetadata;
 
+    /**
+     * @param rootColumnNames Root column names. They are null when called from
+     *     Jet for a sub-relNode and the row metadata aren't needed
+     */
     public PlanCreateVisitor(
-        UUID localMemberId,
-        Map<UUID, PartitionIdSet> partMap,
+        NodeEngine nodeEngine,
```

Comment:
We do not pass NodeEngine for a reason - we do not want to leak internals everywhere, because it breaks decomposition and makes testing harder. Moreover, this refactoring will make merge with master harder, because `PlanCreateVisitor` is already in master.

In general, we should avoid refactorings for now unless there are strong reasons to do so, to minimize wasted work. If refactoring is needed, then it should be performed through `pr -> master -> sql` flow.

Reply:
There are already changes to `PlanCreateVisitor` in `sql` that I needed to change, 3 arguments are added to the constructor: `sql`, `parameterMetadata` and `rootColumnNames`. Will these go to 4.1? If not, then I can do my change against `master`. If yes, then it's probably better to do them to `sql`.

Reply:
`sql` is not really used at the moment, so it can be deleted safely from `PlanCreateVisitor` in the `sql` branch. 
Regarding, `parameterMetadata` and `rootColumnNames`, these changes will be merged to `master` soon as a part of public API PR. 

My point is that changes to `PlanCreateVisitor` appear to be a small refactoring that doesn't change anything from the IMDG perspective. In this case, we should either avoid them altogether or make them through a `master` branch to minimize engineering time spent on conflict resolution. 
But as I mentioned before, not passing NodeEngine to classes when possible was a conscious decision - we do not like such objects to leak without a strong reason, because it promotes bad class design in the longer perspective. AFAIU the reason for this change - to avoid duplication of partition map calculation. In this case, we may just move the method that calculates partition map to a separate utility method and invoke it from Jet. This way we will avoid both code duplication in Jet and `NodeEngine` leak.